script/app-coverage-report.js

```diff
@@ -3,58 +3,89 @@
-    if (key !== 'total') {
-      coverageTable.push({
-        [key]: [
-          `${coverageResults[key].lines.pct}%`,
-          `${coverageResults[key].functions.pct}%`,
-          `${coverageResults[key].statements.pct}%`,
-          `${coverageResults[key].branches.pct}%`,
-        ],
-      });
-    }
+  Object.values(coverageResults).forEach(cov => {
+    const appLocation = `${cov.path.substr(0, cov.path.lastIndexOf('/'))}`;
+
+    coverageTable.push({
+      [appLocation]: [
+        `${cov.lines.pct}%`,
+        `${cov.functions.pct}%`,
+        `${cov.statements.pct}%`,
+        `${cov.branches.pct}%`,
+      ],
+    });
   });
 
   console.log(coverageTable.toString());
 };
 
-const generateCoverage = (rootDir, coverageSummary) =>
-  Object.keys(coverageSummary).reduce((acc, coverageResult) => {
-    const appName = coverageResult.replace(rootDir, '').split(path.sep)[0];
-    // If appName exists, add coverageResult to overall coverage
-    if (appName) {
-      if (acc[appName]) {
-        // Average coverageResult with respective acc segment if present
-        // Each 'seg' represents coverage by line, function, statement, and branch
-        Object.keys(acc[appName]).forEach(seg => {
-          acc[appName][seg].total += coverageSummary[coverageResult][seg].total;
-          acc[appName][seg].covered +=
-            coverageSummary[coverageResult][seg].covered;
-          acc[appName][seg].skipped +=
-            coverageSummary[coverageResult][seg].skipped;
-          acc[appName][seg].pct = (
-            (acc[appName][seg].covered / acc[appName][seg].total) *
-            100
+const generateCoverage = (rootDir, coverageSummary) => {
+  // Get application manifests & entry names
+  const manifests = getAppManifests(path.join(__dirname, '../'));
+  const entryNames = manifests.map(manifest => manifest.entryName);
+
+  // Set initial coverage object
+  const initialCoverage = Object.assign(
+    ...entryNames.map(entryName => ({
+      [entryName]: {
+        path: manifests
+          .find(obj => obj.entryName === entryName)
+          .entryFile.replace(rootDir, ''),
+        lines: { total: 0, covered: 0, pct: 0 },
+        statements: { total: 0, covered: 0, pct: 0 },
+        functions: { total: 0, covered: 0, pct: 0 },
+        branches: { total: 0, covered: 0, pct: 0 },
+      },
+    })),
+  );
+
+  // Iterate through coverages that are under src/applications
+  return Object.keys(coverageSummary)
+    .filter(fullPath => fullPath.replace(rootDir, '').split(path.sep)[0])
+    .reduce((acc, fullCoveragePath) => {
+      // Find coverage if it exists by checking if the path of the coverage result file
+      // includes an application path, e.g. 'vre/chapter36/'
+      const coverageItem = Object.values(acc).find(appCov =>
+        fullCoveragePath
+          .replace(rootDir, '')
+          .includes(
+            `${appCov.path.substring(0, appCov.path.lastIndexOf('/'))}/`,
+          ),
+      );
```

Comment:
The reason for the trailing slash is to verify that `fullCoveragePath` includes a substring of `appCov.path` that is an actual path and not just text. For example, if you remove the trailing slash and run the script, you'll notice that the `auth` application will have app coverage. But this isn't the case because there are no tests in `auth`, it's getting it's coverage from another file in a different app that has the word 'auth' in its path. Adding the trailing slash verifies that the included substring is a path.

Reply:
That makes sense - my mistake.